crates/services/gas_price_service/src/v1/service.rs

```diff
@@ -46,12 +52,11 @@ use tokio::sync::broadcast::{
     Receiver,
 };
 
-use crate::common::utils::Result as GasPriceResult;
-
 /// The service that updates the gas price algorithm.
-pub struct GasPriceServiceV1<L2, Metadata, DA>
+pub struct GasPriceServiceV1<L2, Metadata, DA, U>
 where
     DA: DaBlockCostsSource,
+    U: Clone,
```

Comment:
I don't understand why `U` is bound on `Clone` here and also why it's not bound on `UnrecordedBlocks`